src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+
+        cleanup:
+        isSwDeactivated && lockKeeper.reviveSwitch(swToDeactivate)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(swToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+    }
+
+    @Tidy
+    def "A round trip latency ISL goes down when both switches are DEACTIVATED"() {
+        given: "A round trip latency ISL"
+        Isl roundTripIsl
+        def srcSwToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsl = topology.getRelatedIsls(sw).find {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsl
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable link", false)
+        def dstSwToDeactivate = roundTripIsl.dstSwitch
+
+        when: "Deactivate the src switch"
+        // it takes more time to DEACTIVATE a switch via the 'knockoutSwitch' method on the stage env
+        lockKeeper.knockoutSwitch(srcSwToDeactivate)
+        def isSrcSwDeactivated = true
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip ISL is still DISCOVERED (the system uses round trip latency for ISL alive confirmation)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        when: "Deactivate the dst switch"
+        lockKeeper.knockoutSwitch(dstSwToDeactivate)
+        def isDstSwDeactivated = true
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(dstSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip latency ISL is FAILED (because round_trip_status is not available in DB for current ISL \
+on both switches)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        cleanup:
+        isSrcSwDeactivated && lockKeeper.reviveSwitch(srcSwToDeactivate)
+        isDstSwDeactivated && lockKeeper.reviveSwitch(dstSwToDeactivate)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getSwitch(dstSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+    }
```

Comment:
I'm not sure that I like the scenario. Isn't it better to simultaneously knockout 2 switches? The part that proves that ISL will remain DISCOVERED when 1 side is down is already covered in previous test, while simultaneous knockout may be an edge case (consult with Dmitriy)

Reply:
> Isn't it better to simultaneously knockout 2 switches?

agree, it would be better

test is refactored